test/cli/cli.go

```diff
@@ -0,0 +1,236 @@
+package cli
+
+import (
+	"bytes"
+	"github.com/boson-project/func/test/e2e/utils"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+)
+
+type TestShellCli struct {
+
+	// func binary location (./func)
+	cliBinary string
+	// Run commands from Dir
+	sourceDir string
+	// Testing object
+	t *testing.T
+	// Last Command Execution Status and Results
+	commandExecution *CommandExecution
+
+}
+
+type CommandExecution struct {
+	// Command line
+	CmdLine string
+	// Shell Command Standard output
+	Stdout string
+	// Shell Command Standard Error
+	Stderr string
+	// Additional Error reported by os
+	Error error
+	// Execution Command
+	execCommand *exec.Cmd
+}
+
+func (c *CommandExecution) Dump(t *testing.T) {
+	if c.Stdout != "" {
+		t.Log(c.Stdout)
+	}
+	if c.Stderr != "" {
+		t.Log(c.Stderr)
+	}
+	if c.Error != nil {
+		t.Log(c.Error.Error())
+	}
+}
+
+func (c *CommandExecution) HasError() bool {
+	if c.Error != nil || c.Stderr != "" {
+		return true
+	}
+	return false
+}
+
+func (c *CommandExecution) ErrorDetails() string {
+	if c.Error != nil {
+		return c.Error.Error()
+	}
+	if c.Stderr != "" {
+		return c.Stderr
+	}
+	return ""
+}
+
+
+func NewFuncCli(t *testing.T) *TestShellCli {
+	f := TestShellCli{ t : t}
+	f.cliBinary = os.Getenv("BOSON_FUNC_BIN")
+	if f.cliBinary == "" {
+		t.Log("'func' binary not defined. Please set BOSON_FUNC_BIN envrionment variable prior to run this test")
+		t.FailNow()
+	}
+	_, err := os.Stat(f.cliBinary)
+	if (os.IsNotExist(err)) {
+		t.Logf("'func' binary file not found (%s).", f.cliBinary)
+		t.FailNow()
+	}
+	return &f
+}
+
+func NewKubectlCli(t *testing.T) *TestShellCli {
+	kubectl := TestShellCli{ t : t}
+	kubectl.cliBinary = "kubectl"
+	resp := kubectl.RunSilent("version")
+	if resp.Stdout == "" {
+		t.Log("'kubectl' not found")
```

Comment:
Why not to print the version like with "docker version" ?

Reply:
Yea.. sure! why not :+1:  I'll add it

Reply:
I'm just marking as resolved since there a new pr for e2e